git_test.go

```diff
@@ -1,35 +1,476 @@
 package main
 
 import (
+	"fmt"
+	"io"
+	"log"
 	"os"
+	"path"
 	"testing"
+	"time"
 
+	"github.com/go-git/go-git/v5"
+	"github.com/go-git/go-git/v5/plumbing"
+	"github.com/go-git/go-git/v5/plumbing/object"
 	"github.com/stretchr/testify/assert"
 	"go.opentelemetry.io/otel/attribute"
 )
 
-func TestGit(t *testing.T) {
-	workingDir, err := os.Getwd()
+var workingDir string
+var tempDir string
+
+func init() {
+	wd, err := os.Getwd()
 	if err != nil {
-		t.Error()
+		log.Fatalln("Cannot get current working dir, which is needed by tests")
 	}
 
-	scm := &GitScm{
-		repositoryPath: workingDir,
+	workingDir = wd
+}
+
+// FakeGitRepo downloads Octocat's hello-world repository from Github, providing a simple DSL to add/remove files and commit them into
+// the fake git repository
+type FakeGitRepo struct {
+	repo             *git.Repository
+	divergesFromBase bool
```

Comment:
This attribute will allow us to mark a repository as a change request, or a branch: whenever a new branch is added, or a commit is created, the repo will be marked as diverging from base.